parsers/go.c

```diff
@@ -576,6 +639,8 @@ static void parseFunctionOrMethod (tokenInfo *const token, int scope)
 		if (isType (token, TOKEN_OPEN_CURLY))
 			skipToMatched (token);
 	}
+	else if (receiver_type_token)
```

Comment:
"else" can be removed here and then you can drop the deleteToken() above and reuse the below one for that case too.